widgets/secrets/src/widget.utils.ts

```diff
@@ -1,2 +1,6 @@
 export const translateSecrets = Stage.Utils.getT('widgets.secrets');
 export const translateForm = Stage.Utils.getT('widgets.secrets.form');
+
+export const getSecretProviderOptions = (useSecretProvider: boolean, secretProviderPath: string) => {
+    return useSecretProvider ? { path: secretProviderPath } : undefined;
+};
```

Comment:
## Optional

```suggestion
import type { ProviderOptions } from 'app/widgets/common/secrets/SecretActions';

export const getSecretProviderOptions = (useSecretProvider: boolean, secretProviderPath: string): ProviderOptions => {
    return useSecretProvider ? { path: secretProviderPath } : undefined;
};
```

Reply:
I would even go further and use named parameters here to avoid using boolean argument.

```suggestion

export const getSecretProviderOptions = ({useSecretProvider: boolean, secretProviderPath: string}): ProviderOptions => {
    return useSecretProvider ? { path: secretProviderPath } : undefined;
};
```

Ref.: https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/

Reply:
hmm I need to change the code a bit so the second option would be an empty `{}` not undefined ( https://github.com/cloudify-cosmo/cloudify-stage/pull/2427/commits/10585cbba8c7cb6f4871c7caf838a6f657d3dde2 ✔️ ) - however, I'm not sure how should I implement @qooban's suggestion in that scenario 🤔

Reply:
Hmm... That's interesting 🙂 

Actually looking at that function and it's uses I'm wondering shouldn't we simplify it (then probably we won't have here any issue boolean arguments). 

Right now `doCreate` and `doUpdate` have two optional arguments. 
(BTW Side note: `doCreate` has `providerOptions?: ProviderOptions | null` while `doUpdate` has `providerOptions?: ProviderOptions`. I guess we can drop `null` in the first type.)

What if `providerName` is empty string or undefined - does `providerOptions` matter at all? I guess it shouldn't.
How about doing all that provider options transformations (`string` -> `{ path: string }`) in `SecretActions` and only if provider name is specified - that should allow you to avoid unnecessary conditions and always generate provider options in the same shape:

```
export const getSecretProviderOptions = (path: string): ProviderOptions => ({ path });
```

Reply:
implemented these changes: https://github.com/cloudify-cosmo/cloudify-stage/pull/2427/commits/0d2da85c3c3478c850586ef0f2227aa5a556696f is this what you had in mind @qooban ? :)

Reply:
Yep 👍🏻 Nice, thx.